proof-generator/src/lib.rs

```diff
@@ -0,0 +1,174 @@
+use anyhow::anyhow;
+use coins_bip32::path::DerivationPath;
+use ethers::{
+    prelude::Signer,
+    providers::{Http, Middleware, Provider},
+    signers::{HDPath, Ledger},
+    types::{Signature, H256},
+};
+use serde::{Deserialize, Serialize};
+use std::{
+    convert::TryFrom,
+    fmt::{Debug, Display},
+    fs,
+    io::Write,
+    path::PathBuf,
+    process::{Command, Stdio},
+    str,
+};
+
+/// The options allowed to be provided to the CLI
+#[derive(Debug, Clone)]
+pub struct Options {
+    /// GPG key to sign the proof.
+    pub gpg_key: String,
+    /// Output path of created proof
+    pub output: PathBuf,
+    /// RPC url
+    pub rpc_url: String,
+    /// Account derivation path when using a Ledger hardware wallet.
+    pub ledger_hdpath: Option<DerivationPath>,
+    /// Keystore file containing encrypted private key (default: none).
+    pub keystore: Option<PathBuf>,
+}
+
+/// The proof to be constructed and signed
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Proof {
+    /// GPG signature of predefined proof
+    gpg_sig: String,
+    /// ETH signature of predefined proof
+    eth_sig: Signature,
+}
```

Comment:
I think verifying the proof will require both public keys, so perhaps they should be included in the proof?

Reply:
Hmm yeah I agree, but I think that showing the GPG public key in the proof would be too much.
We could show the GPG fingerprint which can be used to query a keyserver e.g. to obtain the public key..

Reply:
Hmmm, yes you would need at least the key fingerprint then, otherwise there's no way to verify this proof without knowing out-of-band what keys signed it. For ethereum you can include the address.
I also think the message that is being signed should be included in here.

Reply:
Done :+1: